src/engine/platform/scc.cpp

```diff
@@ -0,0 +1,391 @@
+        if (parent->song.resetMacroOnPorta) chan[c.chan].macroInit(parent->getIns(chan[c.chan].ins,DIV_INS_SCC));
+      }
+      chan[c.chan].inPorta=c.value;
+      break;
+    case DIV_CMD_GET_VOLMAX:
+      return 15;
+      break;
+    case DIV_ALWAYS_SET_VOLUME:
+      return 1;
+      break;
+    default:
+      break;
+  }
+  return 1;
+}
+
+void DivPlatformSCC::muteChannel(int ch, bool mute) {
+  isMuted[ch]=mute;
+  if (mute) {
+    rWrite(regBase+15,regPool[regBase+15]&~(1<<ch));
+  } else if (chan[ch].active) {
+    rWrite(regBase+15,regPool[regBase+15]|(1<<ch));
+  }
+}
+
+void DivPlatformSCC::forceIns() {
+  for (int i=0; i<5; i++) {
+    chan[i].insChanged=true;
+    chan[i].freqChanged=true;
+    updateWave(i);
+  }
+}
+
+void* DivPlatformSCC::getChanState(int ch) {
+  return &chan[ch];
+}
+
+DivDispatchOscBuffer* DivPlatformSCC::getOscBuffer(int ch) {
+  return oscBuf[ch];
+}
+
+unsigned char* DivPlatformSCC::getRegisterPool() {
+  return (unsigned char*)regPool;
+}
+
+int DivPlatformSCC::getRegisterPoolSize() {
+  return 225;
+}
+
+void DivPlatformSCC::reset() {
+  memset(regPool,0,225);
+  scc->reset();
+  for (int i=0; i<5; i++) {
+    chan[i]=DivPlatformSCC::Channel();
+    chan[i].std.setEngine(parent);
+    chan[i].ws.setEngine(parent);
+    chan[i].ws.init(NULL,32,255,false);
+    chan[i].vol=15;
+    chan[i].outVol=15;
+    rWrite(regBase+10+i,15);
+  }
+  if (dumpWrites) {
+    addWrite(0xffffffff,0);
+  }
+}
+
+bool DivPlatformSCC::isStereo() {
+  return false;
+}
+
+void DivPlatformSCC::notifyWaveChange(int wave) {
+  for (int i=0; i<5; i++) {
+    if (chan[i].wave==wave) {
+      chan[i].ws.changeWave1(chan[i].wave);
+      updateWave(i);
+    }
+  }
+}
+
+void DivPlatformSCC::notifyInsDeletion(void* ins) {
+  for (int i=0; i<5; i++) {
+    chan[i].std.notifyInsDeletion((DivInstrument*)ins);
+  }
+}
+
+void DivPlatformSCC::poke(unsigned int addr, unsigned short val) {
+  rWrite(addr,val);
+}
+
+void DivPlatformSCC::poke(std::vector<DivRegWrite>& wlist) {
+  for (DivRegWrite& i: wlist) rWrite(i.addr,i.val);
+}
+
+void DivPlatformSCC::setChipModel(bool isplus) {
+  isPlus=isplus;
+}
+
+int DivPlatformSCC::init(DivEngine* p, int channels, int sugRate, unsigned int flags) {
+  parent=p;
+  dumpWrites=false;
+  skipRegisterWrites=false;
+  writeOscBuf=0;
+  for (int i=0; i<5; i++) {
+    isMuted[i]=false;
+    oscBuf[i]=new DivDispatchOscBuffer;
+  }
+  chipClock=COLOR_NTSC;
+  rate=chipClock/16;
+  for (int i=0; i<5; i++) {
+    oscBuf[i]->rate=rate;
+  }
+  if (isPlus) {
+    scc=new k052539_scc_core;
+    regBase=0xa0;
+  } else {
+    scc=new k051649_scc_core;
+    regBase=0x80;
+  }
+  reset();
+  return 5;
+}
+
+void DivPlatformSCC::quit() {
+  for (int i=0; i<5; i++) {
+    delete oscBuf[i];
+  }
+  if (scc!=NULL) {
+    if (isPlus) {
+      delete (k052539_scc_core*)scc;
+    } else {
+      delete (k051649_scc_core*)scc;
+    }
+  }
+}
+
+DivPlatformSCC::~DivPlatformSCC() {
```

Comment:
If this is a no-op, alternatively you could just not define a dtor in your derived class. The fact base class already has a virtual dtor should be enough.

More reading: https://stackoverflow.com/questions/14184341/c-constructor-destructor-inheritance